doc/examples/reconst_dki.py

```diff
@@ -91,13 +91,13 @@
 
 from dipy.core.gradients import gradient_table
 from dipy.data import get_fnames
+from dipy.denoise.localpca import mppca
```

Comment:
Done in https://github.com/dipy/dipy/pull/3197/files#diff-3883fbe25a03fa8e050608f8e3808888811b4e02f06c195b36a76083b331d4b7R94

We should document better how to set up/use the pre-commit hooks. Sorry for that.

In the meantime, the cleanest approach is to (assuming you have a python (virtual) environment set up to work on dipy):
1. Squash all your commits into a single commit (all these changes utitmately do a single thing, i.e. capturing the warnings properly) and follow the DIPY commit subject prefixes: https://github.com/dipy/dipy/blob/50ec2b60209e9945e9dedfd61beaa446f1054a2c/doc/devel/commit_codes.rst
1. Rebase on `master` (assumes `origin` is your remote fork, `upstream` is the dipy org repository):
    ```
    git checkout master
    git fetch upstream
    git merge upstream/master
    git push origin master
    git checkout {your_branch_name}
    git rebase origin/master
    ```
    You may need to solve the conflicts when rebasing on `master`.

    This way we do not introduce unnecessary merge commits.

1. Once that is done, in order to ensure that your local changes comply with the import sorting rules:
    ```
    pip install pre-commit
    pip install ruff
    ruff --fix
    ```
1. Add the changes if files are modified using and do a `git commit --amend --no-edit` if the feature/maint/bugfix/etc. change has been commited with an appropriate message (avoids introducing a new commit to tell about the style change, which is not informative of the feature/maint/bugfix/etc.) ; else, `git commit`.

Hope this helps.

Reply:
Addressed in 3c2b823.